### DISCUSSION TITLE
Include JSON in API-related Dev Docs

### DISCUSSION THREAD

**Discussion by zdmc23** [DiscipleTools/Documentation#39] [More Documentation]
Third-party Integrations with the D.T API send JSON in the body of the request, so let's display the JSON as a "JSON" or "JavaScript" tab in the code block (in addition to existing PHP): https://github.com/GitbookIO/plugin-codetabs

Examples where this would be helpful:

#### Integrations 

For example, `Integromat` (no-code integration service similar to Zapier)

```json
    "url": "{{connection.clientUrl}}/wp-json/dt-posts/v2/contacts/",
    "method": "POST",
    "body": {
        "title": "{{parameters.title}}",
        "contact_phone": [
            {
                "value": "{{parameters.phoneNumber}}"
            }
        ],
        "location_grid": {
            "values": [
                {
                    "value": "{{parameters.location}}"
                }
            ]
        },
        "sources": {
            "values": [
                {
                    "value": "{{parameters.source}}"
                },
                {
                    "value": "{{parameters.sourceCity}}"
                }
            ]
        }
    },
    "response": {
        "output": "{{body}}"
    }
}
```

However, when reviewing the D.T API Dev Docs, the developer is presented with a PHP version that they then need to map from PHP to JSON in their head.  It would just ease the cognitive load if we presented a `Node.js` or `JSON` or `JavaScript` tab that showed the equivalent in JSON format.

```php
$fields = [
  "sources" => [
    "values" => [ 
      [ "value" => "source" ],  //set a value, the value must be predefined in the field options
      [ "value" => "sourceCity", "delete" => true ] //remove existing
    ],
    "force_values" => false // true will set source to the values entries. removing all others
  ]
]
```

#### Testing

It would be helpful when performing any API tests via some HTTP client like `cURL` or `Postman`.

![postman-create-contact](https://user-images.githubusercontent.com/191707/119143220-5babd700-ba15-11eb-8125-bcf468a4b2d3.png)

#### Google Cloud

Here is an example of how Google does this in their GCP API docs (and I have seen similar from Azure and other API providers):

![gcp-code-tabs](https://user-images.githubusercontent.com/191707/119143086-34eda080-ba15-11eb-83e0-2255d3eeb2ed.png)

**Comment by corsacca** [comment]
@zdmc23 can you give an example of where you this would be used?

**Reply by zdmc23** [reply]
Hi @corsacca , sure, I have edited the discussion item to include some examples.  Thx!

**Reply by corsacca** [reply]
Thanks @zdmc23 . I think this would be a great upgrade to the doc!
